editor/plugins/script_editor_plugin.cpp

```diff
@@ -1545,13 +1557,10 @@ bool ScriptEditor::edit(const Ref<Script> &p_script, int p_line, int p_col, bool
 
 	bool open_dominant = EditorSettings::get_singleton()->get("text_editor/files/open_dominant_script_on_scene_change");
 
-	Error err = p_script->get_language()->open_in_external_editor(p_script, p_line >= 0 ? p_line : 0, p_col);
-	if (err == OK)
-		return false;
-	if (err != ERR_UNAVAILABLE)
-		WARN_PRINT("Couldn't open in custom external text editor");
-
-	if (p_script->get_path().is_resource_file() && bool(EditorSettings::get_singleton()->get("text_editor/external/use_external_editor"))) {
+	if ((debugger->get_dump_stack_script() != p_script || debugger->get_debug_with_external_editor()) &&
+			p_script->get_language()->open_in_external_editor(p_script, p_line >= 0 ? p_line : 0, p_col) == OK &&
```

Comment:
I didn't notice this before. This is wrong. `OK` means the language did open the script in an external editor. `ERR_UNAVAILABLE` means it did not open it because it's configured not to do so. Any other error is an actual error. If a language is not supposed to provide this functionality then it does not need to override `open_in_external_editor`, because the default one already returns `ERR_UNAVAILABLE`.
The deleted code was fine. If `open_in_external_editor` returns `OK`, `edit` returns `false` which tells its caller not to switch to the `Script` screen. Otherwise, if the error is not `ERR_UNAVAILABLE`, it's an actual error. Otherwise continue with the normal process: global external editor setting or built-in editor.

Reply:
#10939